src/algokit/cli/tasks/analyze.py

```diff
@@ -0,0 +1,226 @@
+    TEALER_DOT_FILES_ROOT.mkdir(parents=True, exist_ok=True)
+    TEALER_SNAPSHOTS_ROOT.mkdir(parents=True, exist_ok=True)
+
+
+def display_analysis_summary(analysis_results: dict) -> None:
+    """
+    Display the summary of the analysis results.
+
+    Args:
+        analysis_results (dict): Dictionary containing analysis results.
+    """
+    impact_frequency: dict = {}
+    for file_path, result_rows in analysis_results.items():
+        click.echo(f"\nFile: {file_path}\n")
+        for result in result_rows:
+            click.echo(
+                f"Detector: {result[0]}\n"
+                f"Impact: {result[1]}\n"
+                f"Details: {result[2]}\n"
+                f"Execution Paths (#Lines):\n{result[3]}\n"
+            )
+            impact_frequency[result[1]] = impact_frequency.get(result[1], 0) + 1
+    # print summary by impact label
+    click.echo("\nTotal issues:")
+    for impact, frequency in impact_frequency.items():
+        click.echo(f"{impact}: {frequency}")
+
+    click.echo(f"Finished analyzing {len(analysis_results)} files.")
+
+
+def has_template_vars(path: Path) -> bool:
+    """
+    Check if the file contains template variables.
+
+    Args:
+        path (Path): The file path to check.
+
+    Returns:
+        bool: True if template variables are found, False otherwise.
+    """
+    content = path.read_text()
+    return bool(re.search(r"\bTMPL_\w*\b", content))
+
+
+def get_input_files(*, input_paths: tuple[Path], recursive: bool) -> list[Path]:
+    """
+    Get input files based on the input paths and recursive flag.
+
+    Args:
+        input_paths (tuple[Path]): Tuple of input paths.
+        recursive (bool): Flag to indicate recursive search.
+
+    Returns:
+        list[Path]: List of input files.
+    """
+
+    input_files = []
+    for input_path in input_paths:
+        if input_path.is_dir():
+            pattern = "**/*.teal" if recursive else "*.teal"
+            input_files.extend(sorted(input_path.glob(pattern)))
+        else:
+            input_files.append(input_path)
+    return sorted(set(input_files))
+
+
+@click.command(
+    name="analyze",
+    help="Analyze TEAL programs for common vulnerabilities with AlgoKit Tealer integration.",
+)
+@click.argument(
+    "input_paths",
+    nargs=-1,
+    type=click.Path(exists=True, dir_okay=True, file_okay=True, path_type=Path),
+    required=True,
+)
+@click.option(
+    "-r",
+    "--recursive",
+    is_flag=True,
+    cls=MutuallyExclusiveOption,
+    not_required_if=["file"],
+    help="Recursively search for all TEAL files within the provided directory.",
+)
+@click.option(
+    "--force",
+    is_flag=True,
+    help="Force verification without the disclaimer confirmation prompt.",
+)
+@click.option(
+    "-b",
+    "--baseline",
+    is_flag=True,
+    help=(
+        "Exit with a non-zero code if any diffs are identified between the current "
+        "and baseline reports. By default baseline reports are stored in the "
```

Comment:
```suggestion
        "and baseline reports. By default reports are stored in the "
```

Assumption here is that reports are generated every time the command is run. The baseline flag is simply a mechanism for failing when execution doesn't match the last previously generated report.

Also wanted to confirm that a report is not written when the --baseline flag is specified and a diff is detected? This means we can run the exact command again and get the same result.

Reply:
Slightly tweaked the content